beacon-chain/rpc/eth/rewards/handlers.go

```diff
@@ -166,22 +178,318 @@ func (s *Server) BlockRewards(w http.ResponseWriter, r *http.Request) {
+		filteredVals = allVals
+	} else {
+		filteredVals = make([]*precompute.Validator, len(valIndices))
+		for i, valIx := range valIndices {
+			filteredVals[i] = allVals[valIx]
+		}
+	}
+
+	idealRewards := make([]AttReward, 32)
+	idealVals := make([]*precompute.Validator, 32)
+	increment := params.BeaconConfig().EffectiveBalanceIncrement
+	for i := 1; i < 33; i++ {
+		effectiveBalance := uint64(i) * increment
+		idealVals[i-1] = &precompute.Validator{
+			IsActivePrevEpoch:            true,
+			IsSlashed:                    false,
+			CurrentEpochEffectiveBalance: effectiveBalance,
+			IsPrevEpochSourceAttester:    true,
+			IsPrevEpochTargetAttester:    true,
+			IsPrevEpochHeadAttester:      true,
+		}
+		idealRewards[i-1] = &IdealAttestationReward{EffectiveBalance: strconv.FormatUint(effectiveBalance, 10)}
+	}
+	err = attestationsDelta(idealRewards, st, bal, idealVals)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not get attestations delta: " + err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+
+	totalRewards := make([]AttReward, len(valIndices))
+	for i, v := range valIndices {
+		totalRewards[i] = &TotalAttestationReward{ValidatorIndex: strconv.FormatUint(uint64(v), 10)}
+	}
+	err = attestationsDelta(totalRewards, st, bal, filteredVals)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not get attestations delta: " + err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+
+	ir := make([]IdealAttestationReward, len(idealRewards))
+	for i, r := range idealRewards {
+		ir[i] = *r.(*IdealAttestationReward)
+	}
+	tr := make([]TotalAttestationReward, len(totalRewards))
+	for i, r := range totalRewards {
+		tr[i] = *r.(*TotalAttestationReward)
+	}
+	resp := &AttestationRewardsResponse{
+		Data: AttestationRewards{
+			IdealRewards: ir,
+			TotalRewards: tr,
+		},
+	}
+
+	network.WriteJson(w, resp)
+}
+
+func attestationsDelta(
+	rewards []AttReward,
+	beaconState state.BeaconState,
+	bal *precompute.Balance,
+	vals []*precompute.Validator,
+) error {
+	cfg := params.BeaconConfig()
+	prevEpoch := time.PrevEpoch(beaconState)
+	finalizedEpoch := beaconState.FinalizedCheckpointEpoch()
+	increment := cfg.EffectiveBalanceIncrement
+	factor := cfg.BaseRewardFactor
+	baseRewardMultiplier := increment * factor / math.CachedSquareRoot(bal.ActiveCurrentEpoch)
+	leak := helpers.IsInInactivityLeak(prevEpoch, finalizedEpoch)
+
+	// Modified in Altair and Bellatrix.
+	bias := cfg.InactivityScoreBias
+	inactivityPenaltyQuotient, err := beaconState.InactivityPenaltyQuotient()
+	if err != nil {
+		return err
+	}
+	inactivityDenominator := bias * inactivityPenaltyQuotient
+
+	for i, r := range rewards {
+		err = attestationDelta(r, bal, vals[i], baseRewardMultiplier, inactivityDenominator, leak)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func attestationDelta(
```

Comment:
Please rename to `singleAttDelta` or something that is a bit more distinguishable from the plural version

Reply:
https://github.com/prysmaticlabs/prysm/blob/180f310d5393e50c4628786d877bc93b337dde3f/beacon-chain/core/altair/epoch_precompute.go#L256-L293

Seems like the function names are same as here.